buffer/src/main/java/io/netty/buffer/ByteBufUtil.java

```diff
@@ -1242,5 +1242,16 @@ private static boolean isUtf8(ByteBuf buf, int index, int length) {
         return true;
     }
 
+    public static int lengthOfByteOrderMark(ByteBuf buf, Charset charset) {
+        final byte[] bom = "\uFEFF".getBytes(charset);
+        int length = 0;
```

Comment:
If you define i outside of the loop, you can return it directly. and skip the length field.

Reply:
Done